server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java

```diff
@@ -0,0 +1,308 @@
+                "Memory, in bytes, used by the survivor space.",
+                "bytes",
+                () -> new LongWithAttributes(bytesUsedByGCGen(stats.getOrRefresh().getJvm().getMem(), GcNames.SURVIVOR))
+            )
+        );
+
+        metrics.add(
+            registry.registerLongGauge(
+                "es.node.stats.jvm.mem.pools.old.used",
+                "Memory, in bytes, used by the old generation heap.",
+                "bytes",
+                () -> new LongWithAttributes(bytesUsedByGCGen(stats.getOrRefresh().getJvm().getMem(), GcNames.OLD))
+            )
+        );
+
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                "es.node.stats.fs.io_stats.io_time.total",
+                "The total time in millis spent performing I/O operations across all devices used by Elasticsearch.",
+                "milliseconds",
+                () -> new LongWithAttributes(stats.getOrRefresh().getFs().getIoStats().getTotalIOTimeMillis())
+            )
+        );
+    }
+
+    /**
+     * Retrieves the bytes used by a specific garbage collection generation from the provided JvmStats.Mem.
+     *
+     * @param mem  The JvmStats.Mem containing memory pool information.
+     * @param name The name of the garbage collection generation (e.g., "young", "survivor", "old").
+     * @return The number of bytes used by the specified garbage collection generation.
+     */
+    private long bytesUsedByGCGen(JvmStats.Mem mem, String name) {
+        long bytesUsed = 0;
+        for (JvmStats.MemoryPool pool : mem) {
+            if (pool.getName().equals(name)) {
+                bytesUsed = pool.getUsed().getBytes();
+            }
+        }
+        return bytesUsed;
+    }
+
+    /**
+     * Retrieves the current NodeStats for the Elasticsearch node.
+     *
+     * @return The current NodeStats.
+     */
+    private NodeStats getNodeStats() {
+        CommonStatsFlags flags = new CommonStatsFlags(
+            CommonStatsFlags.Flag.Get,
+            CommonStatsFlags.Flag.Search,
+            CommonStatsFlags.Flag.Merge,
+            CommonStatsFlags.Flag.Translog
+        );
+        return nodeService.stats(
+            flags,
+            true,
+            false,
+            false,
+            true,
+            false,
+            true,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false
+        );
+    }
+
+    @Override
+    protected void doStart() {
+        registerAsyncMetrics(registry);
+    }
+
+    @Override
+    protected void doStop() {
+        stats.stopRefreshing();
+    }
+
+    @Override
+    protected void doClose() throws IOException {
+        metrics.forEach(metric -> {
+            try {
+                metric.close();
+            } catch (Exception ignore) {
+                // metrics close() method does not throw Exception
```

Comment:
My gut feeling is that this could be a `logger.warn("metrics close() method should not throw Exception", exception)` so that we have visibility into  unchecked exceptions should they (unexpectedly) occur.  However, I think it's fine to do that as follow up and this should not block merging.